packages/java/endpoint/src/main/java/com/vaadin/hilla/startup/RouteUnifyingServiceInitListener.java

```diff
@@ -39,21 +40,27 @@ public class RouteUnifyingServiceInitListener
 
     private final ClientRouteRegistry clientRouteRegistry;
 
+    private final RouteUtil routeUtil;
+
     private final RouteUnifyingConfigurationProperties routeUnifyingConfigurationProperties;
 
     /**
      * Creates a new instance of the listener.
      *
      * @param clientRouteRegistry
      *            the registry to add the client routes to
+     * @param routeUtil
+     *            the ClientRouteRegistry aware utility for checking if user is
```

Comment:
This JavaDoc and the related method signature make me wonder why we need the `RouteUtil` class in the first place.
That's not related to this PR as the class was already there before, but it exposes a single public method which could be a `ClientRouteRegistry` method, and that would simplify passing `RouteUtil` around.
That's just a thought, not necessarily a request.

Reply:
I think from the cohesion point of view, separating these two classes was a good move, as IMO the `ClientRouteRegistry` should/could remain just a registry and nothing more. Then whoever uses the registry, such as `RouteUtil`, can deal with the content of the registry based on their business.

Though, I think we could at least rename the `RouteUtil` to something more specific, such as `ViewAccessChecker` or `HillaViewAccessChecker` to not get mixed up with the one from Flow.